### PR TITLE
Final Submission HW-1 

### PR DESCRIPTION
**Pull request by mhusseinmidd**
This is the final submission for the HW-1 assignment.

### CODE REVIEW

**Comment by rudeboybert** [comment]
got it. I'll be grading these starting on Thursday.

**Comment by mhusseinmidd** [comment]
?Excellent! Thank you.


Mohamed Hussein
Economics Major, Middlebury College '17
802-349-5132
mhussein@middlebury.edu
________________________________

File: HW-1.Rmd [hunk 1]
```diff
@@ -1,112 +1,259 @@
...
+flights_day <- flights %>% 
+  count(month) %>% 
+  rename(num_flights_day=n)
+
+delayed_flights <- flights %>% 
+  group_by(month) %>% 
+  filter(dep_delay>30) %>% 
+  tally() %>% 
+  rename(num_flights_delayed = n)
+
```
Comment (rudeboybert) [inline HW-1.Rmd]: you could've done lines 61 thru 71 in one shot by defining a new variable `is_delayed` which is equal to 1 if `delay>=30`, then `group_by(month)` then `summarise(prop = mean(is_delayed))`

File: HW-1.Rmd [hunk 2]
```diff
@@ -1,112 +1,259 @@
...
+
+## Question 1:
+
+* Plot a "time series" of the proportion of flights that were delayed by > 30 minutes on each day.  i.e.
+    + the x-axis should be some notion of time
+    + the y-axis should be the proportion.
+* Which seasons did we tend to see the most and least delays of > 30 minutes.
+
+The graphical analysis of the proportion of flights, which were delayed by more than 30 minutes, broken down by months suggests that late spring and early summer are the seasons with most delays. Specifically, the porportoon of delays increases from 11.5% to 13.2% in the period between May and July. Conversly, autumn seems to witness the least amount of delays, fluctuating between 7.5% and 8.5% of all flights.
```
Comment (rudeboybert) [inline HW-1.Rmd]: Why the month resolution? Why not day? Sure, it does simplify things by only having 12 numbers, but people can grasp many more if presented visually. You are losing a lot of information about the variability within month this way.

File: HW-1.Rmd [hunk 3]
```diff
@@ -1,112 +1,259 @@
...
+  filter(!is.na(year)) %>% 
+  group_by(carrier) %>%
+  tally(year<1980) %>% 
+  rename(num_old_planes=n)
+
+Old_planes<-flights_planes %>% 
+  group_by(carrier) %>% 
+  tally() %>% 
+  mutate (Old_planes = round(num_old_planes_per_carrier$num_old_planes/num_planes_per_carrier$num_planes, 3)*100) %>%  
+  arrange(desc(Old_planes))
```
Comment (rudeboybert) [inline HW-1.Rmd]: Again, you could've done this using only a single data frame by creating a variable that you use to count the number of old planes. Ask me more in person if this doesn't make sense.

File: HW-1.Rmd [hunk 4]
```diff
@@ -1,112 +1,259 @@
...
+
+ggplot(Old_planes, aes(x=reorder(carrier, -Old_planes), y=Old_planes)) + geom_bar(stat="identity") + xlab("Carrier") +  ylab("Planes Manifactured <1980 (%) ") + ggtitle("Planes <1980 by Carrier") +  theme(plot.title = element_text(lineheight=.8, face="bold")) + theme(axis.title.y=element_text(margin=margin(0,20,0,0)))
```
Comment (rudeboybert) [inline HW-1.Rmd]: Again, see note about 80 characters.

While this does answer the question, it only gives notion of center i.e. the mean. What about spread within each carrier? One thing I want you guys to get out of this class is that line I said in class from my colleague at Google "Engineers know what a mean is, but statisticians/analysts know what a standard deviation/error is"

File: HW-1.Rmd [hunk 5]
```diff
@@ -1,112 +1,259 @@
...
+
+
+
+## Question 2:
+
+* Some people prefer flying on older planes.  Even though they aren't as nice,
+they tend to have more room.  Which airlines should these people favor?
+
+
+In the 1970s, the airline industry witnessed considerable deregulations that allowed planes to be produced with less legroom; the average seat pitch has since dropped from 35 inces to 31 inches today (http://fortune.com/2015/09/12/airline-seats-shrink/). Consequently, customers who prefer more room should favor flying with Delta, since about 20% of its fleet is composed of airplanes manufactured before 1980.
```
Comment (rudeboybert) [inline HW-1.Rmd]: Very good supplemental notes. This is great adding of value to explain things.

File: HW-1.Rmd [hunk 6]
```diff
@@ -1,112 +1,259 @@
...
+flights_airports<-left_join(flights, airports, by =c("dest"="iata"))
+flights_airports<-flights_airports %>% 
+  filter(carrier=="WN") %>% 
+  group_by(state) %>% 
+  tally() %>% 
+  rename (flights_per_state=n) %>% 
+  arrange (desc(flights_per_state))
+
+q3a<-ggplot(data=flights_airports, aes(x=reorder(state, -flights_per_state), y=flights_per_state)) 
+q3a<-q3a+ geom_bar(stat="identity") + xlab("Destination State") +  ylab("Total no. of Flights") + ggtitle("All Southwest Flights by Destination") +  theme(plot.title = element_text(lineheight=.8, face="bold"))+ theme(axis.title.y=element_text(margin=margin(0,20,0,0)))
```
Comment (rudeboybert) [inline HW-1.Rmd]: See note above about combining fragmented code, unless you are using `q3a` as a base layer for a different plot

File: HW-1.Rmd [hunk 7]
```diff
@@ -1,112 +1,259 @@
...
+
+
+
+
+## Question 3:
+
+* What states did listed Southwest Airlines flights tend to fly to?
+* What states did all Southwest Airlines flights tend to fly to?
+
+It appears that the highest unique number of flights for WN is to Texas. Specifically, WN operates 770 annual unique flights to destinations in Texas, followed by 297 to Florida and 236 to Louisiana. Although the numbers are different, it appears that WN-operated flights tend to fly to the same three destinations, with 17,230 flights to TX, followed by 3,992 to FL and 3,362 to LA. The order, however, is not identical: WN operates more unique flights to New Mexico than Maryland, even though the oveall traffic of flights operated by WN is higher to Maryland than New Mexico.
```
Comment (rudeboybert) [inline HW-1.Rmd]: hmm, southwest is called SOUTHwest. I wonder if that's their historical progression. i.e. started in the south.

File: HW-1.Rmd [hunk 8]
```diff
@@ -1,112 +1,259 @@
...
+
+q4a<-ggplot(firsts, aes(x=temp_first, y=delayed_first)) +geom_point() + geom_smooth(method="lm")
+q4a<-q4a+ xlab("Temprature at planned departure time (F)") +  ylab("Delayed Time")
+q4a+ ggtitle("Temprature vs. Delays") +  theme(plot.title = element_text(lineheight=.8, face="bold"))+ theme(axis.title.y=element_text(margin=margin(0,20,0,0)))
+
+q4b<-ggplot(firsts, aes(x=humidity_first, y=delayed_first)) +geom_point() + geom_smooth(method="lm")
+q4b<-q4b+ xlab("Humidity at planned departure time") +  ylab("Delayed Time")+ ggtitle("Humidity vs. Delays") +  theme(plot.title = element_text(lineheight=.8, face="bold"))+ theme(axis.title.y=element_text(margin=margin(0,20,0,0)))
+q4b
+
+q4c<-ggplot(firsts, aes(x=wind_speed_first, y=delayed_first))+geom_point() + geom_smooth(method="lm")+ ggtitle("Wind Speed vs. Delays") +  theme(plot.title = element_text(lineheight=.8, face="bold"))
```
Comment (rudeboybert) [inline HW-1.Rmd]: I would've used `geom_jitter()` here to spead the points out a little bit. A lot of them are superimposed on top of each other.

File: HW-1.Rmd [hunk 9]
```diff
@@ -1,112 +1,259 @@
...
+
+
+
+
+## Question 4:
+
+* What weather patterns are associated with the biggest departure delays?
+
+
+The three graphs show a modest upwards trend, suggestive of the fact that as the weather condition increases in severity, so does the duration of a flight delayed. That is, for flights that were delayed more than 30 minutes (and less than 3 hours), higher temprature (humidity, or windspeed) is correlated with longer delays.
```
Comment (rudeboybert) [inline HW-1.Rmd]: Regression wasn't part of this assignment, but is one instance where mentioning the confidence interval on the slope parameters would give a sense of how meaningful the supposed positive slopes are.

But what would you do next? Sure, you have these positive relationships. Do they tell the entire picture?

File: HW-1.Rmd [hunk 10]
```diff
@@ -1,112 +1,259 @@
...
+
+
+
+
+
+## Question 5:
+
+* I want to know what proportionately regions (NE, south, west, midwest) each carrier flies to from Houston in the month of July.  Consider the `month()` function from the `lubridate` package.
+
+The graph below shows the proportion of regions each carrier flies to in the month of July. Notice that the grey color, unreported in the legend, reflects flights for which there was no information on the region of destination.
```
Comment (rudeboybert) [inline HW-1.Rmd]: Anything surprising in the graph? I found it funny that AA, which is based in nearby Dallas, only flew to southern destinations from Houston, and nowhere else. Meaning, if I'm part of AAdvantage, and I want to go anywhere other than the south, I need to transfer at least once.

**Pull request closed without merging**